src/DTNtupleEnvironmentFiller.cc

```diff
@@ -108,17 +115,32 @@ void DTNtupleEnvironmentFiller::fill(const edm::Event & ev)
 	      break;
 	    }
 	}
+      if(debug) std::cout<<"[DTNupleEnvironmentFiller] actualPU = "<<m_actualPileUp<<" | truePU = "<<m_truePileUp<<std::endl;
     }
 
-  auto lumiScalers = conditionalGet<OnlineLuminosityRecord>(ev, m_lumiScalerToken, "OnlineLuminosityRecord");
-
-  if (lumiScalers.isValid()) 
+  auto lumiScalers     = conditionalGet<OnlineLuminosityRecord>(ev, m_lumiScalerToken, "OnlineLuminosityRecord");     // 2018-2022
+  auto lumiScalers2017 = conditionalGet<LumiScalersCollection>(ev,  m_lumiScalerToken_2017, "LumiScalersCollection"); // 2017
+  double instLumi_2018=0, instLumi_2017=0;
+  int    onlinePileUp_2018=0, onlinePileUp_2017=0;
+  if(debug) std::cout<<"[DTNupleEnvironmentFiller] lumiScalers.isValid() = "<<lumiScalers.isValid()<<" lumiScalers2017.isValid() = "<<lumiScalers2017.isValid()<<std::endl;
+  if (lumiScalers.isValid()) // 2018-2022
     {
-      m_instLumi = lumiScalers->instLumi();
+      instLumi_2018 = lumiScalers->instLumi();      // 2018-2022
+      onlinePileUp_2018 = lumiScalers->avgPileUp(); // 2018-2022
+      std::cout<<"[DTNupleEnvironmentFiller] [lumiScalers 2018-2022] instLumi = "<<instLumi_2018<<" E30 cm-2s-1 | avgPU = "<<onlinePileUp_2018<<std::endl;
     }
+  if (lumiScalers2017.isValid()) // 2017
+    {
+      instLumi_2017 = lumiScalers2017->begin()->instantLumi(); // 2017
+      onlinePileUp_2017 = lumiScalers2017->begin()->pileup();  // 2017
+      std::cout<<"[DTNupleEnvironmentFiller] [lumiScalers 20XX-2017] instLumi = "<<instLumi_2017<<" E30 cm-2s-1 | avgPU = "<<onlinePileUp_2017<<std::endl;
+    }
+  if(instLumi_2017>0) { m_instLumi = instLumi_2017; m_onlinePileUp = onlinePileUp_2017; }
+  if(instLumi_2018>0) { m_instLumi = instLumi_2018; m_onlinePileUp = onlinePileUp_2018; }
```

Comment:
The logic is a little fishy, no?
Only one between `lumiScalers`  and `lumiScalers2017` should be valid, correct?

may be:
```c++
if (lumiScalers.isValid()) {
// fill m_* here
} else if (lumiScalers2017.isValid()) {
// fill m_* here
}
```

?